packages/core/contracts/EditionNFTContract.test.ts

```diff
@@ -54,125 +54,310 @@ describe('EditionNFTContract', () => {
+      expect(onChainEdition.size).toEqual(edition.limit);
+      expect(onChainEdition.isClosed).toBe(true);
+    });
   });
 
-  it('should mint 3 edition 1 NFTs into default bucket', async () => {
+  describe('Edition with limit and partial mint', () => {
+    let edition: EditionResult;
+
+    it('should create the edition', async () => {
+      const editionInput = {
+        limit: 5,
+        metadata: {
+          name: 'Edition 2',
+          description: 'This is the second edition.',
+          thumbnail: 'edition-2.jpeg',
+        },
+      };
+
+      edition = await client.send(contract.createEdition(editionInput));
+
+      const onChainEdition = await client.query(contract.getEdition(edition.id));
+
+      expect(onChainEdition.id).toEqual(edition.id);
+      expect(onChainEdition.limit).toEqual(edition.limit);
+      expect(onChainEdition.size).toEqual(0);
+    });
+
     const count = 3;
 
-    await client.send(contract.mintNFTs({ editionId: edition1.id, count }));
+    it('should mint part of the edition', async () => {
+      const mintedNFTs = await client.send(contract.mintNFTs({ editionId: edition.id, count }));
 
-    const onChainEdition = await client.query(contract.getEdition(edition1.id));
+      for (const nft of mintedNFTs) {
+        expect(nft.editionId).toEqual(edition.id);
+      }
 
-    expect(onChainEdition.id).toEqual(edition1.id);
-    expect(onChainEdition.size).toEqual(edition1.size);
-    expect(onChainEdition.count).toEqual(count);
-  });
+      const onChainEdition = await client.query(contract.getEdition(edition.id));
 
-  it('should mint remaining 2 edition 1 NFTs into default bucket', async () => {
-    const count = 2;
+      expect(onChainEdition.id).toEqual(edition.id);
+      expect(onChainEdition.limit).toEqual(edition.limit);
+      expect(onChainEdition.size).toEqual(count);
+    });
 
-    await client.send(contract.mintNFTs({ editionId: edition1.id, count }));
+    it('should close the edition early', async () => {
+      await client.send(contract.closeEdition(edition.id));
 
-    const onChainEdition = await client.query(contract.getEdition(edition1.id));
+      const onChainEdition = await client.query(contract.getEdition(edition.id));
 
-    expect(onChainEdition.id).toEqual(edition1.id);
-    expect(onChainEdition.size).toEqual(edition1.size);
-    expect(onChainEdition.count).toEqual(edition1.size);
-  });
+      expect(onChainEdition.id).toEqual(edition.id);
+      expect(onChainEdition.limit).toEqual(edition.limit);
+      expect(onChainEdition.size).toEqual(count);
+    });
 
-  it('should fail to mint more than edition size', async () => {
-    await expect(async () => {
-      await client.send(contract.mintNFTs({ editionId: edition1.id, count: 5 }));
-    }).rejects.toThrow();
+    it('should fail to mint after closing', async () => {
+      await expect(async () => {
+        await client.send(contract.mintNFTs({ editionId: edition.id, count: 5 }));
```

Comment:
Test would be cleaner if you attempt to mint 1 NFT.  Since the limit is 5 and the size here is 3, calling `mintNFTs` with `count: 5` could be erroring because you would theoretically exceed the limit, rather than because the edition is closed.
```suggestion
        await client.send(contract.mintNFTs({ editionId: edition.id, count: 1 }));
```

Reply:
Great catch, thank you!